### PR TITLE
Update README.md

### PR DESCRIPTION
**Pull request by damiencaselli**
Removes every occurrences of `ObjectController` since it is deprecated.

### CODE REVIEW

**Comment by bcardarella** [comment]
In this case the validation paths should be changed from `firstName` to `model.firstName`

**Comment by damiencaselli** [comment]
Ah stupid me, half-way done work :8ball: 
I'll correct this PR.

**Comment by damiencaselli** [comment]
@bcardarella I've updated the tests examples, but since a `Controller` can validate its own properties without referring to its model, the correction does not seem necessary.

In the examples, we may want to explicitly say a controller always validates its model properties for better comprehension, or use component, which would leverage the ambiguity.

File: README.md [hunk 1]
```diff
@@ -61,7 +61,7 @@ to the property. If you pass `true` then the property itself will be
 seen as a validatable object.
 
 ```javascript
-export default Ember.ObjectController.extend({
+export default Ember.Controller.extend({
   validations: {
     firstName: {
```
Comment (bcardarella) [inline README.md]: this should be:

```js
validations: {
  'model.firstName': {
    presence: true
  }
}
```

Comment (damiencaselli) [inline README.md]: I did some rebasing here, referred commit changes are still part of the PR.

**Pull request closed without merging**